plugin/net/driver.go

```diff
@@ -100,22 +100,16 @@ func (driver *driver) EndpointInfo(req *api.EndpointInfoRequest) (*api.EndpointI
 func (driver *driver) JoinEndpoint(j *api.JoinRequest) (*api.JoinResponse, error) {
 	driver.logReq("JoinEndpoint", j, fmt.Sprintf("%s:%s to %s", j.NetworkID, j.EndpointID, j.SandboxKey))
 
-	local, err := createAndAttach(j.EndpointID, WeaveBridge, 0)
-	if err != nil {
+	name, peerName := vethPair(j.EndpointID)
+	if _, err := weavenet.CreateAndAttachVeth(name, peerName, WeaveBridge, 0, nil); err != nil {
```

Comment:
Yes, that's part of "refactoring the current plugin code to be closer to the script"; since we have just created the veth, and we don't tell anyone else about it until we get to `JoinResponse` below, we should delete it on error otherwise it leaks.